client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -74,7 +74,9 @@ const CheckoutThankYou = React.createClass( {
 		this.redirectIfThemePurchased();
 
 		if ( this.props.receipt.hasLoadedFromServer && this.hasPlanOrDomainRegistration() ) {
-			this.props.refreshSitePlans( this.props.selectedSite.ID );
+			this.props.refreshSitePlans( this.props.selectedSite );
+		} else if ( shouldFetchSitePlans( this.props.sitePlans, this.props.selectedSite ) ) {
+			this.props.fetchSitePlans( this.props.selectedSite );
 		}
```

Comment:
That's probably personal preference but I would have switched clauses:

```javascript
if ( shouldFetchSitePlans( this.props.sitePlans, this.props.selectedSite ) ) {
  this.props.fetchSitePlans( this.props.selectedSite );
} else if ( this.props.receipt.hasLoadedFromServer && this.hasPlanOrDomainRegistration() ) {
  this.props.refreshSitePlans( this.props.selectedSite );
}
```

That's way it feels more logical (to me) since you first load the site plans before refreshing them: 

```
if ( site plans not yet loaded or loading ) {
  load site plans
} else if ( receipt loaded and is for domain or plan ) {
  refresh site plans
}
```